### ISSUE TITLE
Non-owner referenced child resources

### ISSUE THREAD

**Issue by Samze** [vmware-labs/reconciler-runtime#83]
Hey folks,

As part of the reconciler we're writing we may need to create some resources that won't have an owner reference, e.g. cluster scoped or resources in a different namespace. The ChildReconciler implementation automatically sets the owner reference so isn't suitable. 

Any thoughts on the best way to handle this within the framework?  Things that spring to mind is a sync reconciler that manages these resources and the finalizers to ensure cleanup, or we could implement a more specific subreconciler to handle these non-namespaced child resources more generally.

Interested in your thoughts

Cheers,
Sam

**Comment by scothis** [comment]
This is a gap in the current ChildReconciler. There are a couple pieces to consider in a potential solution:
1. tell the ChildReconciler to not automatically add the owner reference (and likewise discover children based on the owner ref)
   
   The recently added `OurChild` func on the ChildReconciler can combine with a flag to opt out of the owner ref injection and filtering. This bit is trivial.
   
2. manage the cleanup of children when the parent resource is deleted
   
   As you mentioned, this is the role of finalizers. At the moment, we don't have explicit support for either adding finalizers or removing finalizers from a resource pending deletion. It's something we should add support for. This will require a thoughtful approach.

**Comment by gmrodgers** [comment]
hey @scothis, I'm encountering the same issue when working with reconciler-runtime. To get around it I've done something similar to what @Samze suggested, namely using a SyncReconciler and then doing finalizer logic manually myself.  I did have to make two changes though to a local version of reconciler runtime:
- Allowing the test factory for ObjectMeta to update the Finalizers
- On a delete (timestamp set), the SyncReconciler was running the `Sync` function and was immediately returned so I commented this out...

Would these two changes be something you'd accept into the main branch (in separate PRs perhaps)? Would be great to here thoughts on this
